src/generator.c

```diff
@@ -0,0 +1,50 @@
+#define _GNU_SOURCE // asprintf
+#include <stdio.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+#include "action.h"
+#include "generator.h"
+#include "menu.h"
+#include "util.h"
+
+nm_menu_item_t **nm_generator_do(nm_generator_t *gen, size_t *sz_out) {
+    NM_LOG("generator: running generator (%s) (%s) (%d) (%p)", gen->desc, gen->arg, gen->loc, gen->generate);
+
+    char *err;
+    size_t sz;
```

Comment:
Probably a good idea (unless we *want* to catch nasty generators with random segfaults? ^^).

Reply:
That's probably a good idea, since it's an easy mistake to make.

Note: We currently catch misusages of `err_out` with segfaults (it should be set to NULL upon success [rather than being left in an undefined state], and should be a malloc'd string on error [rather than a literal or alloca'd string]). I'll leave that one as is since everything will work if the macros are used properly.